From f7c6df674f7c5556c4a58d03310eb84e1d4f4676 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Fri, 22 Mar 2019 19:55:07 +0200
Subject: [PATCH] update-kernel: include associated firmware files to modloop

Some drivers reference only the .bin but require additional
files such as .$board.txt or .clm_blob. Include all files
in modloop that might match. ref #9549
---
 update-kernel.in | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/update-kernel.in b/update-kernel.in
index eeb0716..0079182 100644
--- a/update-kernel.in
+++ b/update-kernel.in
@@ -278,10 +278,10 @@ mkdir -p $MODLOOP/modules/firmware
 find $ROOT/lib/modules -type f -name "*.ko" | xargs modinfo -F firmware | sort -u | while read FW; do
 	if [ -e "$ROOT/lib/firmware/$FW" ]; then
 		install -pD $ROOT/lib/firmware/$FW $MODLOOP/modules/firmware/$FW
-		# include nvram files if firmware is needed.
-		if [ -e "$ROOT/lib/firmware/${FW%.*}.txt" ]; then
-			install -pD $ROOT/lib/firmware/${FW%.*}.txt $MODLOOP/modules/firmware/${FW%.*}.txt
-		fi
+		# copy also all potentially associated files
+		for _file in "$ROOT"/lib/firmware/"${FW%.*}".*; do
+			install -pD "$_file" "$MODLOOP/modules/firmware/${_file#*/lib/firmware/}"
+		done
 	fi
 done
 
-- 
2.21.0

